### PR TITLE
Adds a name for the Chef's pet Goat

### PR DESCRIPTION
**Pull request by SabreML**
## What Does This PR Do

The Chef's goat in the freezer now has a custom name and description. It also has `unique_pet = 1`, in order to prevent renaming via collars.

## Why It's Good For The Game

Most of the other animals have names. e.g. Betsy, Commander Clucky.
Surely the Goat should have one too, considering how iconic it's become.

## Images of changes

Old:
Capture

New:
Capture

## Changelog
:cl:
add: The Chef's goat is now named Pete
/:cl:

### CODE REVIEW

**Comment by Samman166** [comment]
AMAZING I LOVE IT

**Comment by AffectedArc07** [comment]
I dont agree with naming an IC simple animal based off of an OOC character, especially when its based on an (admittedly awful) meme.

**Comment by SabreML** [comment]
Willing to change the name to 'Pete' if it's too meme-y, since that's apprently their unofficial name.
I'll wait a while for more feedback either way.

*(To be honest I just wanted to get this out there, since I can't be bothered to wait until April 1st.)*

**Comment by Mitchs98** [comment]
I thought it was named Pete this whole time myself 🤔

**Comment by SabreML** [comment]
> I thought it was named Pete this whole time myself 🤔

Just 'a goat' previously.

Seems like even the name 'Pete' is based on an OOC character; An old contributor for /tg/, 'Petethegoat'.

**Comment by AffectedArc07** [comment]
> Seems like even the name 'Pete' is based on an OOC character; An old contributor for /tg/, 'Petethegoat'.

Pete is an actually normal name for something. You wouldnt call a goat Kyet in the real world unless you were really out of your mind. It fits for an online screen name, not for the name of an animal.

**Comment by SabreML** [comment]
> Yeah...no.

Eh, worth a try.

They should still get a name though, any suggestions?

**Comment by Spacemanspark** [comment]
Pete works.

**Comment by KasparoVy** [comment]
Cuban Pete, the king of the rhumba beat

**Review by Kyep — CHANGES_REQUESTED** [review]
Objection. 
1) Please don't name IC mobs after OOC ckeys. Especially without even asking the person first.
2) This gives the goat meme too much prominence.
3) If you're going to make the chef's goat special, a slight color tint might be good, since that is visible from a distance, and makes them much more recognizable.

**Comment by SabreML** [comment]
> 1. Please don't name IC mobs after OOC ckeys. Especially without even asking the person first.

Yeah, sorry about that. Should I ask Petethegoat for permission then or what?

**Comment by taukausanake** [comment]
Named or not, I'm pretty sure the chef's goat is going to be called Pete either way. It's one of those unwritten laws that everyone knows but not why.

And yeah, Kyet is a weird name but I feel like one of those Northern Europe countries would name a goat something weird like that

**Comment by SabreML** [comment]
> Named or not, I'm pretty sure the chef's goat is going to be called Pete either way. It's one of those unwritten laws that everyone knows but not why.

Having only even played on Paradise and CM, I've honestly never heard of 'Pete' until now.
I probably would've gone with that name to begin with, had I known about it.

**Comment by fludd12** [comment]
My only complaint is that Pete is being named through a variable edit and not making a subtype, to be honest!

**Comment by SabreML** [comment]
> My only complaint is that Pete is being named through a variable edit and not making a subtype, to be honest!

I figured it would be cleaner to just edit it on the map rather than making a subtype.

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 1]
```diff
@@ -30989,7 +30989,11 @@
 	},
 /area/hydroponics)
 "bfE" = (
-/mob/living/simple_animal/hostile/retaliate/goat,
+/mob/living/simple_animal/hostile/retaliate/goat{
+	desc = "Pete, the Chef's pet goat from the Caribbean. Not known for their pleasant disposition.";
```
Comment (Kyep) [inline _maps/map_files/cyberiad/cyberiad.dmm:30993]: Don't varedit things on the map like this.
Instead, create a subtype in the code and have the map file reference this instead.
This makes it easier, in future, to update the object without new map edits.

Comment (SabreML) [reply _maps/map_files/cyberiad/cyberiad.dmm:30993]: Didn't consider the 'map edit' side, I'll change it back to a subtype.

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 2]
```diff
@@ -99,6 +99,11 @@
 		H.visible_message("<span class='warning'>[src] takes a big chomp out of [H]!</span>", "<span class='userdanger'>[src] takes a big chomp out of your [NB.name]!</span>")
 		NB.droplimb()
 
+/mob/living/simple_animal/hostile/retaliate/goat/chef
+	name = "Pete"
+	desc = "Pete, the Chef's pet goat from the Caribbean. Not known for their pleasant disposition."
+	unique_pet = 1
```
Comment (SteelSlayer) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:105]:
```suggestion
	unique_pet = TRUE
```

**Review by Kyep — COMMENTED** [review]
Objection retracted.

**Comment by Pixelzilla** [comment]
I always figured the goat had no name since his main reason for existing is generally to be butchered. Same with the pig. Betsy and Clucky are both animals that provide constant resources (milk and eggs which can make more chickens).

**Review by Dumbdumn5 — APPROVED** [review]
Praise Pete 👍

**Review by farie82 — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**